pkg/ipamd/ipamd.go

```diff
@@ -1186,6 +1374,79 @@ func (c *IPAMContext) verifyAndAddIPsToDatastore(eni string, attachedENIIPs []*e
 	return seenIPs
 }
 
+// verifyAndAddPrefixesToDatastore updates the datastore with the known Prefixes. Prefixes who are out of cooldown gets added
+// back to the datastore after being verified against EC2.
+func (c *IPAMContext) verifyAndAddPrefixesToDatastore(eni string, attachedENIIPs []*ec2.Ipv4PrefixSpecification, needEC2Reconcile bool) map[string]bool {
+	var ec2VerifiedAddresses []*ec2.Ipv4PrefixSpecification
+	seenIPs := make(map[string]bool)
+	for _, privateIPv4 := range attachedENIIPs {
+		strPrivateIPv4 := aws.StringValue(privateIPv4.Ipv4Prefix)
+		log.Debugf("Check in coolddown Found prefix %s", strPrivateIPv4)
+
+		// Check if this Prefix was recently freed
+		_, ipv4CidrPtr, err := net.ParseCIDR(strPrivateIPv4)
+		if err != nil {
+			log.Debugf("Failed to parse so continuing with next prefix")
+			continue
+		}
+		found, recentlyFreed := c.reconcileCooldownCache.RecentlyFreed(strPrivateIPv4)
+		if found {
+			log.Debugf("found in cooldown")
+			if recentlyFreed {
+				log.Debugf("Reconcile skipping IP %s on ENI %s because it was recently unassigned from the ENI.", strPrivateIPv4, eni)
+				continue
+			} else {
+				if needEC2Reconcile {
```

Comment:
nit: we should split the loop into two. 
1. first go though the attachedENIIPs to see filter out attachedENIIPs that is not RecentlyFreed
2. then if the filtered list contains items and needEC2Reconcile, we do the APICall
3. then we go through the filtered list and do the rest of the logic.

Reply:
Thanks Yang, this makes sense. We should do this even for the secondary IP function.